src/cryptography/hazmat/primitives/twofactor/utils.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import unicode_literals
+
+import base64
+
+from six.moves.urllib.parse import quote, urlencode
+
+
+def generate_uri(hotp, type_name, account_name, issuer, extra_parameters):
```

Comment:
This utility function is meant for consumption by the existing classes rather than by an end user so I'd prefer it to be `_generate_uri`